multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -108,6 +111,8 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 	r.mapLock.Lock()
 	defer r.mapLock.Unlock()
```

Comment:
Please change "r.mapLock.Unlock()" to be after line 147 (line 142 in the old code).